test/new-e2e/utils/e2e/stack_definition.go

```diff
@@ -100,3 +102,32 @@ func AgentStackDef(vmParams []ec2params.Option, agentParameters ...agentparams.O
 		},
 	)
 }
+
+type DockerEnv struct {
+	Docker *client.Docker
+}
+
+// DockerStackDef creates a stack definition for Docker.
+//
+// See [dockerparams.Params] for available options for params.
+//
+// [dockerparams.Params]: https://pkg.go.dev/github.com/DataDog/test-infra-definitions@main/components/datadog/agent/dockerparams#Params
+func DockerStackDef(params ...dockerparams.Option) *StackDefinition[DockerEnv] {
+	return EnvFactoryStackDef(
+		func(ctx *pulumi.Context) (*DockerEnv, error) {
+			vm, err := ec2vm.NewUnixEc2VM(ctx)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
We should have a `NewDockerEc2VM` to ensure we use the proper Docker AMI

Reply:
This is updated in a later commit.